### PR TITLE
Support symfony 5.3

### PR DESCRIPTION
**Pull request by Punk-UnDeaD**
#44 support symfony 5.3

### CODE REVIEW

File: src/DependencyInjection/CompilerPass/MiddlewareCompilerPass.php [hunk 1]
```diff
@@ -29,7 +29,9 @@ public function process(ContainerBuilder $container): void
         $defaultMiddlewares = $container->getParameter('baldinof_road_runner.middlewares.default');
 
         $middlewaresToRemove = [];
-        if (!$container->hasDefinition('session')) {
+        if (!$container->hasDefinition('session')
+            && empty(array_reduce($container->getExtensionConfig('framework'), 'array_merge', [])['session']['enabled'])
```
Comment (Baldinof) [inline src/DependencyInjection/CompilerPass/MiddlewareCompilerPass.php]: Maybe we can always rely on the framework config? And never check the container here?

Also, why using `array_reduce` / `array_merge` ? Can it works with 
```php
if (!$container->getExtensionConfig('framework')['session']['enabled'])) {
```

Or if we are not sure the `session.enabled` is defined: 

```php

$sessionEnabled = $container->getExtensionConfig('framework')['session']['enabled']) ?? false;

if(!$sessionEnabled) {
```

**Review by Baldinof — COMMENTED** [review]
Hi!

Good catch :)

File: src/DependencyInjection/CompilerPass/MiddlewareCompilerPass.php (hunk 1 above)
Comment (Punk-UnDeaD) [reply src/DependencyInjection/CompilerPass/MiddlewareCompilerPass.php]: $container->getExtensionConfig('framework') hasn't 'session' key

look my screenshot https://idiod.video/ji2wd2.png

Comment (Baldinof) [reply src/DependencyInjection/CompilerPass/MiddlewareCompilerPass.php]: Ahah, clearly it's not what I expected to be returned from `getExtensionConfig()` :D

It returns configs before it's been parsed by [this class](https://github.com/symfony/symfony/blob/cfb1016a7c1fac23fb8bbe95f7737ecbdffa3b5e/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php), so the `session.enabled` key could not be defined.

Also if the `session.enabled` is not defined, it means for symfony that the session is enabled, see
https://github.com/symfony/symfony/blob/cfb1016a7c1fac23fb8bbe95f7737ecbdffa3b5e/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L614
and https://github.com/symfony/symfony/blob/cfb1016a7c1fac23fb8bbe95f7737ecbdffa3b5e/src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php#L280

I think a another approach would be to keep checking the existence of a service (or a parameter) that is defined when the session is enabled and not deprecated (and available from 4.4 to 5.3), maybe `session.handler` ([4.4](https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L952) and [5.3](https://github.com/symfony/symfony/blob/5.3/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L1100)), or the `session.metadata.storage_key` parameter.

What do you think?

Comment (Punk-UnDeaD) [reply src/DependencyInjection/CompilerPass/MiddlewareCompilerPass.php]: good point, 'session.enabled' default true if 'session' config exist

so we can simple check `$container->hasDefinition('session.factory')`

**Comment by Baldinof** [comment]
Thank you :)

**Pull request merged**